Build failure: rspamd

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix-build -A rspamd`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/326101735

### Relevant log output

```console
In file included from /build/source/src/libmime/mime_string.cxx:18:
/nix/store/k3xmbic33n5gbd68ss6ll4rsc74mv9jx-doctest-2.5.0/include/doctest/doctest.h: In instantiation of 'static void doctest::detail::filldata<T, typename std::enable_if<((! doctest::detail::has_insertion_operator<T>::value) && doctest::detail::is_container<T>::value)>::type>::fill(std::ostream*, const T&) [with T = rspamd::mime::basic_mime_string<char>; std::ostream = std::basic_ostream<char>]':
/nix/store/k3xmbic33n5gbd68ss6ll4rsc74mv9jx-doctest-2.5.0/include/doctest/doctest.h:1086:22:   required from 'void doctest::detail::filloss(std::ostream*, const T&) [with T = rspamd::mime::basic_mime_string<char>; std::ostream = std::basic_ostream<char>]'
 1086 |     filldata<T>::fill(stream, in);
      |     ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
/nix/store/k3xmbic33n5gbd68ss6ll4rsc74mv9jx-doctest-2.5.0/include/doctest/doctest.h:1099:12:   required from 'doctest::String doctest::detail::toStream(const T&) [with T = rspamd::mime::basic_mime_string<char>]'
 1099 |     filloss(stream, in);
      |     ~~~~~~~^~~~~~~~~~~~
/nix/store/k3xmbic33n5gbd68ss6ll4rsc74mv9jx-doctest-2.5.0/include/doctest/doctest.h:1107:24:   required from 'static doctest::String doctest::detail::StringMakerBase<true>::convert(const T&) [with T = rspamd::mime::basic_mime_string<char>]'
 1107 |         return toStream(in);
      |                ~~~~~~~~^~~~
/nix/store/k3xmbic33n5gbd68ss6ll4rsc74mv9jx-doctest-2.5.0/include/doctest/doctest.h:1144:35:   required from 'doctest::String doctest::toString(const T&) [with T = rspamd::mime::basic_mime_string<char>; typename std::enable_if<(! detail::should_stringify_as_underlying_type<T>::value), bool>::type <anonymous> = true]'
 1144 |     return StringMaker<T>::convert(value);
      |            ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/nix/store/k3xmbic33n5gbd68ss6ll4rsc74mv9jx-doctest-2.5.0/include/doctest/doctest.h:1290:13:   required from 'doctest::String doctest::detail::stringifyBinaryExpr(const L&, const char*, const R&) [with L = rspamd::mime::basic_mime_string<char>; R = char [1]]'
 1290 |     return (DOCTEST_STRINGIFY(lhs)) + op + (DOCTEST_STRINGIFY(rhs));
      |             ^
/nix/store/k3xmbic33n5gbd68ss6ll4rsc74mv9jx-doctest-2.5.0/include/doctest/doctest.h:2000:5:   required from 'decltype (((void)((declval<L>() == declval<R>())), (doctest::detail::Result)(<brace-enclosed initializer list>()))) doctest::detail::Expression_lhs<L>::operator==(R&&) [with R = const char (&)[1]; L = const rspamd::mime::basic_mime_string<char>&; decltype (((void)((declval<L>() == declval<R>())), (doctest::detail::Result)(<brace-enclosed initializer list>()))) = doctest::detail::Result]'
 2000 |     DOCTEST_DO_BINARY_EXPRESSION_COMPARISON(==, " == ", DOCTEST_CMP_EQ)
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/libmime/mime_string.cxx:31:4:   required from here
   31 |                         CHECK(st == "");
      |                                     ^~
/nix/store/k3xmbic33n5gbd68ss6ll4rsc74mv9jx-doctest-2.5.0/include/doctest/doctest.h:1277:32: error: passing 'const rspamd::mime::basic_mime_string<char>' as 'this' argument discards qualifiers [-fpermissive]
 1277 |         for (auto it = in.begin(); it != in.end(); ++it) {
      |                        ~~~~~~~~^~
In file included from /build/source/src/libmime/mime_string.cxx:19:
/build/source/src/libmime/mime_string.hxx:526:21: note:   in call to 'rspamd::mime::basic_mime_string<T, Allocator, Functor>::iterator rspamd::mime::basic_mime_string<T, Allocator, Functor>::begin() [with CharT = char; Allocator = std::allocator<char>; Functor = fu2::abi_400::detail::function<fu2::abi_400::detail::config<false, true, fu2::capacity_default>, fu2::abi_400::detail::property<true, false, int(int)> >; iterator = rspamd::mime::iterator<rspamd::mime::basic_mime_string<char>, false>]'
  526 |         inline auto begin() noexcept -> iterator
      |                     ^~~~~
/nix/store/k3xmbic33n5gbd68ss6ll4rsc74mv9jx-doctest-2.5.0/include/doctest/doctest.h:1277:48: error: passing 'const rspamd::mime::basic_mime_string<char>' as 'this' argument discards qualifiers [-fpermissive]
 1277 |         for (auto it = in.begin(); it != in.end(); ++it) {
      |                                          ~~~~~~^~
/build/source/src/libmime/mime_string.hxx:536:21: note:   in call to 'rspamd::mime::basic_mime_string<T, Allocator, Functor>::iterator rspamd::mime::basic_mime_string<T, Allocator, Functor>::end() [with CharT = char; Allocator = std::allocator<char>; Functor = fu2::abi_400::detail::function<fu2::abi_400::detail::config<false, true, fu2::capacity_default>, fu2::abi_400::detail::property<true, false, int(int)> >; iterator = rspamd::mime::iterator<rspamd::mime::basic_mime_string<char>, false>]'
  536 |         inline auto end() noexcept -> iterator
      |                     ^~~
/nix/store/k3xmbic33n5gbd68ss6ll4rsc74mv9jx-doctest-2.5.0/include/doctest/doctest.h:1278:31: error: passing 'const rspamd::mime::basic_mime_string<char>' as 'this' argument discards qualifiers [-fpermissive]
 1278 |             if (it != in.begin()) {
      |                       ~~~~~~~~^~
/build/source/src/libmime/mime_string.hxx:526:21: note:   in call to 'rspamd::mime::basic_mime_string<T, Allocator, Functor>::iterator rspamd::mime::basic_mime_string<T, Allocator, Functor>::begin() [with CharT = char; Allocator = std::allocator<char>; Functor = fu2::abi_400::detail::function<fu2::abi_400::detail::config<false, true, fu2::capacity_default>, fu2::abi_400::detail::property<true, false, int(int)> >; iterator = rspamd::mime::iterator<rspamd::mime::basic_mime_string<char>, false>]'
  526 |         inline auto begin() noexcept -> iterator
      |                     ^~~~~
make[2]: *** [src/CMakeFiles/rspamd-server.dir/build.make:2305: src/CMakeFiles/rspamd-server.dir/libmime/mime_string.cxx.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1848: src/CMakeFiles/rspamd-server.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

### Additional context

Not sure what happened here, last `nixos-unstable` revision built just fine.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.23, NixOS, 26.05 (Yarara), 26.05.20260418.b12141e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.34.6`
 - nixpkgs: `/nix/store/qmqfjcv57sha8g976zqycdkvairb461j-source`

### Notify maintainers

@avnik @fpletz @nlewo

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/rs/rspamd/package.nix
+++ b/pkgs/by-name/rs/rspamd/package.nix
@@ -3,7 +3,7 @@
   lib,
   fetchFromGitHub,
   cmake,
-  doctest,
+  # doctest,
   fmt,
   perl,
   glib,
@@ -59,7 +59,7 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
   buildInputs = [
-    doctest
+    # doctest
     fmt
     glib
     openssl
@@ -94,7 +94,7 @@ stdenv.mkDerivation (finalAttrs: {
     "-DENABLE_BLAS=${if withBlas then "ON" else "OFF"}"
     "-DENABLE_FASTTEXT=ON"
     "-DENABLE_JEMALLOC=ON"
-    "-DSYSTEM_DOCTEST=ON"
+    "-DSYSTEM_DOCTEST=OFF" # https://github.com/rspamd/rspamd/issues/5994
     "-DSYSTEM_FMT=ON"
     "-DSYSTEM_XXHASH=ON"
     "-DSYSTEM_ZSTD=ON"

